### PR TITLE
RISC-V (riscv64gc) Build Fix Survey

### PR DESCRIPTION
**Pull request by Icenowy**
Topic Description
-----------------

This survey fixes FTBFSes met in riscv64 initial building process.

Package(s) Affected
-------------------

```
linux+api
expat
dpkg
c-ares
nghttp2
e2fsprogs
gtk-doc
funcsigs
util-macros
x11-proto
xcb-proto
soupsieve
nspr
elfutils
libcap
tk
gcc
rustc
lz4
gtest
unzip
rustc
libvdpau
ruby
libcl
mesa
gpm
vim
freeglut
llvm
librsvg
ijs
ghostscript
ijs
breathe
scandir
sphinxcontrib-htmlhelp
sphinxcontrib-serializinghtml
zope-interface
cryptography
twisted
xmltoman
ftjam
mpdecimal
argyllcms
xorg-server
python-3
sane-backends
libxkbcommon
aspell
gsettings-desktop-schemas
mariadb
tidyhtml
jemalloc
mariadb
httpd
libsodium
argon2
graphviz
hyphen
vulkan
pulseaudio
smpeg
celt
libftdi
soxr
ibus
v4l-utils
xvidcore
chrpath
lttng-ust
numactl
libbluray
xf86-video-amdgpu
xf86-video-ati
x11-base
gstreamer-1-0
libcanberra
fprintd
arping
hwloc
openal-soft
qt-5
scons
serf
libutf8proc
m17n-db
wireless-tools
js-78
xfsprogs
openmpi
thin-provisioning-tools
hdparm
kernel-tools
dhcp
ppp
tlp
snowball
libcork
libipset
rhash
libapt-pkg-perl
nfs-utils
debianutils
fcron
gettext
cargo-audit
```

Security Update?
----------------
No

Build Order
-----------

Any order should be okay because this survey shouldn't introduce incompatible change.

Architectural Progress
----------------------

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

Secondary Architectural Progress
--------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

Post-Merge Secondary Architectural Progress
-------------------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

File: extra-devel/ftjam/autobuild/prepare [hunk 1]
```diff
@@ -0,0 +1,5 @@
+abinfo "Setting CCFLAGS environment variable, which is what Jam wants ..."
+export CCFLAGS="$CFLAGS"
+
+abinfo "Removing strip command from Jamfile ..."
+sed -i 's/strip/:/g' Jamfile
```
Comment (MingcongBai) [inline extra-devel/ftjam/autobuild/prepare]: Absolute path here.

File: extra-doc/ijs/autobuild/defines [hunk 2]
```diff
@@ -3,6 +3,8 @@ PKGSEC=doc
 PKGDEP="glibc"
 BUILDDEP="ghostscript"
 PKGDES="Implements a protocol for transmission of raster page images"
+NOSTATIC=0
+NOLTO=1
```
Comment (MingcongBai) [inline extra-doc/ijs/autobuild/defines:7]: Why?

File: extra-i18n/aspell/02-dict/build [hunk 3]
```diff
@@ -1,8 +1,9 @@
-wget https://repo.anthonos.org/aosc-repacks/aspell-dicts-$DICTVER.tar.xz
-tar xf aspell-dicts-$DICTVER.tar.xz
+abinfo "Extracting dicts tarball ..."
+tar xf ../aspell-dicts.tar.xz
```
Comment (MingcongBai) [inline extra-i18n/aspell/02-dict/build]: Absolute path here.

File: extra-i18n/aspell/02-dict/build [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
 
-pushd aspell-dicts-$DICTVER
+pushd aspell-dicts-*
```
Comment (MingcongBai) [inline extra-i18n/aspell/02-dict/build]: Ditto.

File: extra-i18n/ibus/02-ibus/build [hunk 5]
```diff
@@ -4,4 +4,4 @@ make -C src DESTDIR="$PKGDIR" uninstall
 make -C bindings DESTDIR="$PKGDIR" uninstall
 make DESTDIR="$PKGDIR" uninstall-pkgconfigDATA
 
-rm -r "$PKGDIR"/usr/lib/python2.7/site-packages/gi
+rm -r "$PKGDIR"/usr/lib/python3.8/site-packages/gi
```
Comment (MingcongBai) [inline extra-i18n/ibus/02-ibus/build]: Use `$ABPY3VER` here.

File: extra-libs/c-ares/autobuild/patch [hunk 6]
```diff
@@ -0,0 +1,2 @@
+abinfo "Changing flag for disabling debug info to adding debug info ..."
+sed -i 's/-g0/-ggdb/g' m4/cares-compilers.m4
```
Comment (MingcongBai) [inline extra-libs/c-ares/autobuild/patch]: Absolute path here.

File: extra-libs/hyphen/autobuild/beyond [hunk 7]
```diff
@@ -1,3 +1,3 @@
-make -C build DESTDIR="$PKGDIR" install-libLTLIBRARIES
-make -C build DESTDIR="$PKGDIR" install-binSCRIPTS
-make -C build DESTDIR="$PKGDIR" install-includeHEADERS
+make -C abbuild DESTDIR="$PKGDIR" install-libLTLIBRARIES
```
Comment (MingcongBai) [inline extra-libs/hyphen/autobuild/beyond]: Absolute path here, and for the next two lines.

File: extra-libs/libcork/autobuild/patch [hunk 8]
```diff
@@ -0,0 +1,5 @@
+abinfo "Dropping -Werror ..."
+sed -i 's/-Werror//g' CMakeLists.txt
```
Comment (MingcongBai) [inline extra-libs/libcork/autobuild/patch]: Absolute path here. Ditto below.

File: extra-libs/libipset/autobuild/prepare [hunk 9]
```diff
@@ -0,0 +1,2 @@
+abinfo "Dropping -Werror ..."
+sed -i 's/-Werror//g' CMakeLists.txt
```
Comment (MingcongBai) [inline extra-libs/libipset/autobuild/prepare]: Absolute path here.

File: extra-libs/libutf8proc/autobuild/build [hunk 10]
```diff
@@ -4,7 +4,7 @@ cmake .. ${CMAKE_DEF} ${CMAKE_AFTER}
 make
 
 install -Dm644 "$SRCDIR"/utf8proc.h "$PKGDIR"/usr/include/libutf8proc/utf8proc.h
-install -Dm644 libutf8proc.so.2.3.1 \
+install -Dm755 libutf8proc.so.2.3.1 \
```
Comment (MingcongBai) [inline extra-libs/libutf8proc/autobuild/build]: Verbosity here.

File: extra-libs/qt-5/autobuild/build [hunk 11]
```diff
@@ -94,7 +95,7 @@ mkdir -pv "$SRCDIR"/../qtwebkit-${PKGVER:9}-alpha4/build
         -DLIB_INSTALL_DIR=/usr/lib
     abinfo "Appending -latomic linkage to build.ninja ..."
     sed -e '/LINK_LIBRARIES = / s/$/ -latomic/' \
-        -i "$SRCDIR"/qtwebkit-build/build.ninja
+        -i build.ninja
```
Comment (MingcongBai) [inline extra-libs/qt-5/autobuild/build:98]: Absolute path here.

File: extra-libs/qt-5/autobuild/prepare [hunk 12]
```diff
@@ -6,6 +11,12 @@ sed -e "s|^\(QMAKE_CXXFLAGS_RELEASE.*\)|\1 ${CXXFLAGS}|" \
 sed -e "s|^\(QMAKE_LFLAGS_RELEASE.*\)|\1 ${LDFLAGS}|" \
     -i "$SRCDIR"/qtbase/mkspecs/common/g++-unix.conf
 
+abinfo "Using -pthread instead of -lpthread for threading support ..."
+for i in qtbase/mkspecs/common/linux.conf qtbase/src/corelib/configure.json
```
Comment (MingcongBai) [inline extra-libs/qt-5/autobuild/prepare]: Absolute path here.

File: extra-libs/serf/autobuild/prepare [hunk 13]
```diff
@@ -0,0 +1,6 @@
+abinfo "BLFS: Patching SConstruct file to remove RPATH and support Python-3 ..."
+sed -i "/Append/s:RPATH=libdir,::"          SConstruct &&
```
Comment (MingcongBai) [inline extra-libs/serf/autobuild/prepare]: Absolute path here, also consider merging your sed's.

File: extra-network/dhcp/autobuild/prepare [hunk 14]
```diff
@@ -1 +1,2 @@
-sed '/^CFLAGS="$CFLAGS/ s/INGS"/INGS -D_GNU_SOURCE"/' -i configure
+sed '/^CFLAGS="$CFLAGS/ s/INGS"/INGS -D_GNU_SOURCE -fcommon"/' -i configure
+
```
Comment (MingcongBai) [inline extra-network/dhcp/autobuild/prepare]: Trailing blank line here.

File: extra-tcl/tk/autobuild/build [hunk 15]
```diff
@@ -1,16 +1,25 @@
 cd unix
+abinfo "Configuring ..."
 ./configure --prefix=/usr \
```
Comment (MingcongBai) [inline extra-tcl/tk/autobuild/build]: Absolute path here.

File: extra-utils/debianutils/autobuild/prepare [hunk 16]
```diff
@@ -0,0 +1,2 @@
+abinfo "Running po4a ..."
+( cd po4a && po4a --no-backups po4a.conf )
```
Comment (MingcongBai) [inline extra-utils/debianutils/autobuild/prepare]: Absolute path here.

File: extra-utils/gpm/autobuild/patches/LFS-gpm-1.20.7-consolidated-1.patch [hunk 17]
```diff
@@ -0,0 +1,116 @@
+Submitted by: Pierre Labastie <pierre dot labastie at neuf dot fr>
```
Comment (MingcongBai) [inline extra-utils/gpm/autobuild/patches/LFS-gpm-1.20.7-consolidated-1.patch]: Patch name convention non-compliant.

File: extra-utils/hdparm/autobuild/patches/hdparm-9.58-prevent-strip.patch [hunk 18]
```diff
@@ -0,0 +1,20 @@
+diff -Naur hdparm-9.58.orig/Makefile hdparm-9.58/Makefile
```
Comment (MingcongBai) [inline extra-utils/hdparm/autobuild/patches/hdparm-9.58-prevent-strip.patch]: Patch name convention non-compliant.

File: extra-utils/unzip/autobuild/prepare [hunk 19]
```diff
@@ -0,0 +1,2 @@
+abinfo "Dropping linking flag to strip symbols ..."
+sed -i '/LF2     =/d' unix/Makefile
```
Comment (MingcongBai) [inline extra-utils/unzip/autobuild/prepare]: Absolute path here.

File: extra-web/httpd/autobuild/patches/httpd-2.4.46-fix-envvars-location.patch [hunk 20]
```diff
@@ -0,0 +1,30 @@
+diff -Naur httpd-2.4.46.orig/support/Makefile.in httpd-2.4.46/support/Makefile.in
```
Comment (MingcongBai) [inline extra-web/httpd/autobuild/patches/httpd-2.4.46-fix-envvars-location.patch]: Patch name convention non-compliant.

File: base-admin/dpkg/01-update-alternatives/build [hunk 21]
```diff
@@ -1,3 +1,6 @@
+abinfo "Generating autotools files ..."
+autoreconf -fi
+
 abinfo "Building DPKG and installing to a fake rootfs ..."
 ./configure \
```
Comment (MingcongBai) [inline base-admin/dpkg/01-update-alternatives/build]: Absolute path here.

File: base-admin/dpkg/01-update-alternatives/build [hunk 22]
```diff
@@ -1,3 +1,6 @@
+abinfo "Generating autotools files ..."
+autoreconf -fi
```
Comment (MingcongBai) [inline base-admin/dpkg/01-update-alternatives/build]: Verbosity here.

File: extra-admin/xfsprogs/autobuild/build [hunk 23]
```diff
@@ -2,6 +2,9 @@ abinfo 'Storing $SRCDIR value and unsetting $SRCDIR ...'
 export ABSRCDIR="$SRCDIR"
 unset SRCDIR
 
+abinfo "Replacing config.{guess,sub} ..."
+cp -v /usr/share/automake-1.16/config.{guess,sub} .
```
Comment (MingcongBai) [inline extra-admin/xfsprogs/autobuild/build]: Absolute path here. @liushuyu You might want to expose config.* replacement as a function as well.

File: extra-admin/xfsprogs/autobuild/build [hunk 24]
```diff
@@ -2,6 +2,9 @@ abinfo 'Storing $SRCDIR value and unsetting $SRCDIR ...'
...
+
 abinfo "Configuring xfsprogs ..."
 ./configure \
```
Comment (MingcongBai) [inline extra-admin/xfsprogs/autobuild/build]: Absolute path here.

File: extra-database/mariadb/autobuild/defines [hunk 25]
```diff
@@ -34,10 +34,9 @@ CMAKE_AFTER="-DCMAKE_AR=/usr/bin/gcc-ar \
              -DWITH_SSL=system \
              -DWITH_PCRE=bundled \
              -DWITH_LIBWRAP=OFF \
-             -DWITH_JEMALLOC=ON \
+             -DWITH_JEMALLOC=yes \
```
Comment (MingcongBai) [inline extra-database/mariadb/autobuild/defines:37]: Huh?

File: extra-i18n/ibus/01-libibus/patches/ibus-1.5.25-fix-non-py3-default-py.patch [hunk 26]
```diff
@@ -0,0 +1,9 @@
+diff -Naur ibus.old/engine/gensimple.py ibus/engine/gensimple.py
```
Comment (MingcongBai) [inline extra-i18n/ibus/01-libibus/patches/ibus-1.5.25-fix-non-py3-default-py.patch]: Patch name convention non-compliant.

File: extra-libs/freeglut/autobuild/patches/LFS-freeglut-3.2.1-gcc10_fix-1.patch [hunk 27]
```diff
@@ -0,0 +1,53 @@
+Submitted By:            Pierre Labastie <pierre dot labastie at neuf dot fr>
```
Comment (MingcongBai) [inline extra-libs/freeglut/autobuild/patches/LFS-freeglut-3.2.1-gcc10_fix-1.patch]: Patch name convention non-compliant.

File: extra-libs/js-78/autobuild/patches/debian-Add-riscv64-support.patch [hunk 28]
```diff
@@ -0,0 +1,147 @@
+From: William Grant <wgrant@ubuntu.com>
```
Comment (MingcongBai) [inline extra-libs/js-78/autobuild/patches/debian-Add-riscv64-support.patch]: Patch name convention non-compliant.

File: extra-libs/qt-5/autobuild/patches/debian-support_riscv64.patch.deferred [hunk 29]
```diff
@@ -0,0 +1,61 @@
+Description: add support for riscv64 architecture
```
Comment (MingcongBai) [inline extra-libs/qt-5/autobuild/patches/debian-support_riscv64.patch.deferred]: Patch name convention non-compliant.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-database/mariadb/autobuild/defines (hunk 25 above)
Comment (Icenowy) [reply extra-database/mariadb/autobuild/defines:37]: The CMakeLists seems to be accepting yes/no

Comment (Icenowy) [reply extra-database/mariadb/autobuild/defines:37]: see https://github.com/MariaDB/server/blob/ddcb242b3cd2e95bcf4c950bffaafc0775db45ee/cmake/jemalloc.cmake#L4

Comment (MingcongBai) [reply extra-database/mariadb/autobuild/defines:37]: I see, that's all good. Just making sure.

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**